src/Http/ServerRequest.php

```diff
@@ -257,28 +254,21 @@ protected function _setConfig(array $config): void
             $config['environment']['REQUEST_METHOD'] = 'GET';
         }
 
-        $this->_environment = $config['environment'];
         $this->cookies = $config['cookies'];
 
-        if (isset($config['uri']) && $config['uri'] instanceof UriInterface) {
+        if (isset($config['uri'])) {
+            if (!$config['uri'] instanceof UriInterface) {
+                throw new Exception('The `uri` key must be an instanceof Psr\Http\Message\UriInterface.');
```

Comment:
```suggestion
                throw new Exception('The `uri` key must be an instance of ' . UriInterface::class);
```

Should we use the generated name?